scene/2d/visible_on_screen_notifier_2d.cpp

```diff
@@ -31,13 +31,33 @@
 #include "visible_on_screen_notifier_2d.h"
 
 #ifdef DEBUG_ENABLED
+Dictionary VisibleOnScreenNotifier2D::_edit_get_state() const {
+	Dictionary state = Node2D::_edit_get_state();
+	state["rect"] = rect;
+	return state;
+}
+
+void VisibleOnScreenNotifier2D::_edit_set_state(const Dictionary &p_state) {
+	ERR_FAIL_COND(p_state.is_empty() || !p_state.has("rect"));
+	set_rect(p_state["rect"]);
+	Node2D::_edit_set_state(p_state);
+}
+
+void VisibleOnScreenNotifier2D::_edit_set_rect(const Rect2 &p_edit_rect) {
+	set_rect(p_edit_rect);
+}
+
 Rect2 VisibleOnScreenNotifier2D::_edit_get_rect() const {
 	return rect;
 }
 
 bool VisibleOnScreenNotifier2D::_edit_use_rect() const {
 	return true;
 }
+
+bool VisibleOnScreenNotifier2D::_edit_is_selected_on_click(const Point2 &p_point, double p_tolerance) const {
+	return show_rect && is_visible() && CanvasItem::_edit_is_selected_on_click(p_point, p_tolerance);
```

Comment:
Sorry, I meant it's selectable when `show_rect` is disabled.
Invisible CanvasItems are automatically filtered out.